spec/index.bs

```diff
@@ -4835,56 +4837,115 @@ dictionary GPUComputePassDescriptor : GPUObjectDescriptorBase {
+        packed block of **three 32-bit unsigned integer values (12 bytes total)**, given in the same
+        order as the arguments for {{GPUComputePassEncoder/dispatch()}}. For example:
+
+        ```js
+        let dispatchIndirectParameters = new Uint32Array(3);
+        dispatchIndirectParameters[0] = x;
+        dispatchIndirectParameters[1] = y;
+        dispatchIndirectParameters[2] = z;
+        ```
 
         <div algorithm="GPUComputePassEncoder.dispatchIndirect">
             **Called on:** {{GPUComputePassEncoder}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPUComputePassEncoder/dispatchIndirect(indirectBuffer, indirectOffset)">
-                indirectBuffer:
-                indirectOffset:
+                |indirectBuffer|: Buffer containing the [=indirect dispatch parameters=].
+                |indirectOffset|: Offset in bytes into |indirectBuffer| where the dispatch data begins.
             </pre>
 
-            **Returns:** void
+            **Returns:** `void`
+
+            Issue the following steps on the [=Device timeline=] of |this|.{{GPUObjectBase/[[device]]}}:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - It is [$valid to dispatch$] with |this|.
+                        - |indirectBuffer| is [$valid to use with$] |this|.
+                        - |indirectBuffer|.{{GPUBuffer/[[usage]]}} contains {{GPUBufferUsage/INDIRECT}}.
+                        - |indirectOffset| + sizeof([=indirect dispatch parameters=]) &le;
+                            |indirectBuffer|.{{GPUBuffer/[[size]]}}.
+                        - |indirectOffset| is a multiple of 4.
 
-            Issue: Describe {{GPUComputePassEncoder/dispatchIndirect()}} algorithm steps.
+                        Issue: Describe additional {{GPUComputePassEncoder/dispatchIndirect()}} algorithm steps.
+                    </div>
+                1. Add |indirectBuffer| to the [=usage scope=] as {{GPUBufferUsage/INDIRECT}}.
+            </div>
         </div>
 </dl>
 
+<div algorithm>
+    To determine if it's <dfn abstract-op>valid to dispatch</dfn> with {{GPUComputePassEncoder}} |encoder|
+    run the following steps:
+
+    1. If any of the following conditions are unsatisfied return `false`:
+        <div class=validusage>
+            - |encoder|.{{GPUComputePassEncoder/[[pipeline]]}} is not `null`.
+            - Let |pipelineLayout| be |encoder|.{{GPUComputePassEncoder/[[pipeline]]}}.{{GPUPipelineBase/[[layout]]}}.
+            - For each pair of ({{GPUIndex32}} |index|, {{GPUBindGroupLayout}} |bindGroupLayout|) in
+                |pipelineLayout|.{{GPUPipelineLayout/[[bindGroupLayouts]]}}.
+                - Let |bindGroup| be |encoder|.{{GPUProgrammablePassEncoder/[[bind_groups]]}}[|index|].
+                - |bindGroup| is not `null`.
```

Comment:
Here's another case where perhaps it would be best if we redid the style to use "must be" instead of "is". Here we mix "let" statements (sentences) and "is" conditions (not really sentences) and it is kind of confusing to read IMO.

I think this convinces me that a consistent "must" style would probably be better, despite the awkwardness of "must be greater than or equal to" or "must be &ge;". No urgency to figure that out and go through and update everything (it's a lot of updates...), but for places like this where it's clearer, like here, I think we should use "must" now.